source/common/http/conn_manager_impl.h

```diff
@@ -199,6 +215,16 @@ class ConnectionManagerConfig {
   virtual bool useRemoteAddress() PURE;
 
   /**
+   * @return ForwardClientCertType the configuration of how to forward the client certs.
+   */
+  virtual ForwardClientCertType forwardClientCert() PURE;
+
+  /**
+   * @return ClientCertDetailsType the details of the client cert to forward to the next hop.
+   */
+  virtual const std::list<ClientCertDetailsType> setClientCertDetails() PURE;
```

Comment:
The name of this method is strange. From its name, it seems to imply it is mutating (i.e. it sets something), but it takes no arguments. Can you explain how this is supposed to be used more clearly? Also, it returns a list of `ClientCertDetailsType`, which is also a bit confusing, since this is basically just a Subject/SAN enum, why a list of this?